Mathlib/LinearAlgebra/UnitaryGroup.lean

```diff
@@ -198,6 +198,19 @@ set_option linter.uppercaseLean3 false in
 
 end UnitaryGroup
 
+section specialUnitaryGroup
+
+variable (n) (α : Type v) [CommRing α] [StarRing α] [Invertible (2 : α)] {A : Matrix n n α}
```

Comment:
It seems that this typeclass assumption is unused.
```suggestion
variable (n) (α : Type v) [CommRing α] [StarRing α] {A : Matrix n n α}
```

Reply:
I added it because the people in [this](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Odd.20Fintype) and in [this](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Banach.20Tarski.20Paradox.20in.20Lean) zulip thread said that we need to assume that 2 is invertible in order to define it as the kernel of the determinant

Reply:
But currently I don't think that you are actually assuming that 2 is invertible. You just have an unused typeclass assumption floating around.

Reply:
How can I specify this assumption then, should I add it directly in front of the definition?

Reply:
That would work, except that the linter will complain that the assumption is unused :)

Personally, I think that it's simplest just to drop the assumption for now, mention in the docstring that the definition isn't quite right in characteristic 2, and add the assumption later whenever it's needed in a proof.

Reply:
Ok, if this will go through...